Bearded.Utilities/Input/InputManager.cs

```diff
@@ -48,13 +47,11 @@ public void Update(bool windowIsActive)
             {
                 keyboardEvents.Update();
                 mouseEvents.Update();
-                keyboardState.Update();
-                mouseState.Update();
```

Comment:
Yes, sadly OpenTK decided to switch to updating the keyboard state and mouse state themselves. There is a `GetSnapshot` method, but even that doesn't really solve our problems (and it's not very performant).

I dug into the OpenTK source to see where this is updated, and it's part of the `ProcessEvents` call in `NativeWindow`. (Except for MouseState is updated twice?? Following up with OpenTK to check this is intentional.) So you're right that this would break in the way we do threading in awgraphics/TD.

I'm open to suggestions on how to fix this without completely rewriting our input stack. I'd rather not keep this library on an OpenTK pre-release until we can do so. Or perhaps this should be the focus of our upcoming programming session 🙈

Reply:
I'm fine going ahead with this, but it might break TD input entirely.

Reply:
Did you test that? Can we test that?

Reply:
We don't have any easy example projects to use.